plugins/ndpi-dummy/ndpi.c

```diff
@@ -0,0 +1,84 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/* License note: While this "glue" code to the nDPI library is GPLv2,
+ * nDPI is itself LGPLv3 which is known to be incompatible with the
+ * GPLv2. */
+
+#include "suricata-common.h"
+#include "suricata-plugin.h"
+#include "util-debug.h"
+
+#include "flow-callbacks.h"
+#include "flow-storage.h"
+
+static FlowStorageId flow_storage_id = { .id = -1 };
+
+static void *FlowStorageAlloc(unsigned int size)
+{
+    SCLogNotice("Allocating nDPI flow storage, size=%d", size);
+    return NULL;
+}
+
+static void FlowStorageFree(void *ptr)
+{
+    SCLogNotice("De-allocating nDPI flow storage");
+    int *dummy_storage = ptr;
+    SCLogNotice("%d", *dummy_storage);
+    SCFree(ptr);
+}
+
+static void OnFlowInit(Flow *f, const Packet *p)
```

Comment:
I guess I'd expect a OnFlowDestruct too, although perhaps we don't need it directly here, as the FlowStorage API knowns how to destroy the user data.

Reply:
Yeah, for completeness should probably add it. Not needed for storage cleanup using the storage API, and someone else uses the flow event for their private cleanup, but it does feel missing.